api/src/main/java/org/openmrs/module/sync2/api/model/configuration/SyncConfiguration.java

```diff
@@ -1,19 +1,25 @@
 package org.openmrs.module.sync2.api.model.configuration;
 
+import java.util.ArrayList;
 import java.util.Objects;
 
 public class SyncConfiguration {
 
     private GeneralConfiguration general;
     private SyncMethodConfiguration push;
     private SyncMethodConfiguration pull;
+    private WhitelistConfiguration whitelist;
 
-    public SyncConfiguration() { }
+    public SyncConfiguration() {
+        whitelist = getDefaultWhitelistConfiguration();
```

Comment:
Should we do the same with other fields?

Reply:
The whitelist is a functionality which isn't obligatory. That's why a user does not have to provide configuration for it. The rest of configurations support core functionalities, so they should be configurated.

Reply:
Okay, that makes sense.